### PR TITLE
Feature/rr 946 date range

### PR DESCRIPTION
**Pull request by au2**
If this PR fixes a bug, you _must_ add test cases representative of the bug.

Please refer to [Tettra](https://app.tettra.co/teams/amida/pages/amida-pull-request-and-code-review-guide) for PR review guidelines.

#### How should this be manually tested?

Add dateRange as specified in swagger to enableWhen and test functionality. Enable when logic should be in and source question should be of type date.

### CODE REVIEW

**Comment by unprofessional** [comment]
For the sake of my work in zipRange, why don't you use `in-date-range` for your logic so I can use `in-zip-range` and have some more self-description for our logic operators?

**Comment by au2** [comment]
done

**Comment by unprofessional** [comment]
I'm primary, with either @jsachs  or @sirctseb  as secondary code reviewer (whoever gets to it first). Functional will be done by @Arghast as instructed by me (to get it set up, etc).

File: models/dao/user-survey.dao.js [hunk 1]
```diff
@@ -29,6 +29,29 @@ const comparators = {
 
         return !_.isEqual(answers, ruleAnswers);
     },
+    'in-date-range': function (answers, ruleAnswers) {
+        if (!(answers && answers.length)) {
+            return false;
+        }
+        if (answers.length !== 1 || ruleAnswers.length !== 1) {
+            return false; // not yet implemented
```
Comment (jsachs) [inline models/dao/user-survey.dao.js]: Is there an issue number we can reference here for what the implementation will look like?

**Comment by unprofessional** [comment]
@Arghast I've left comments in https://github.com/amida-tech/recruitment-registry-client/pull/228 regarding what you need to do to test this backend PR. You're actually testing this backend for functuonality by using that frontend branch -- please note your findings here in this PR.

File: models/dao/user-survey.dao.js (hunk 1 above)
Comment (au2) [reply models/dao/user-survey.dao.js]: RR-962 is added.

**Comment by au2** [comment]
@unprofessional and/or @jsachs is this good to go now?

**Comment by unprofessional** [comment]
screen shot 2018-03-27 at 2 05 46 pm

I get a problem on the `migrations.spec` test

**Comment by unprofessional** [comment]
Oh, it was on my end. Fixed. Looks good.

**Pull request merged**